src/test/ui/local-ident.stderr

```diff
@@ -0,0 +1,24 @@
+error[E0308]: mismatched types
+  --> $DIR/local-ident.rs:15:16
+   |
+15 |     let x: X = S;
+   |                ^ expected struct `Mod2::X`, found struct `Mod1::S`
+   |
+   = note: expected type `X`
+              found type `S`
+
+error[E0308]: mismatched types
+  --> $DIR/local-ident.rs:16:28
+   |
+16 |     let y: Option<usize> = Ok(2);
+   |                            ^^^^^ expected enum `std::option::Option`, found enum `std::result::Result`
```

Comment:
This is an open question: Do we want to show the fully qualified path here? I don't think length is as much of an issue here as this label doesn't print the type arguments, keeping the line relatively short. Alternatively it might be jarring to have two different things being shown here...